x/sockopt/is_sending_bytes_linux.go

```diff
@@ -0,0 +1,42 @@
+//go:build linux
+
+package sockopt
+
+import (
+	"net"
+
+	"golang.org/x/sys/unix"
+)
+
+func isSocketFdSendingBytes(fd int) (bool, error) {
+	tcpInfo, err := unix.GetsockoptTCPInfo(int(fd), unix.IPPROTO_TCP, unix.TCP_INFO)
+	if err != nil {
+		return false, err
+	}
+
+	// 1 == TCP_ESTABLISHED, but for some reason not available in the package
+	if tcpInfo.State != 1 {
```

Comment:
Why is this relevant? Won't `tcpInfo.Notsent_bytes` be zero if the connection is not established?

Also, we won't have the conenction object if it's not established.

Reply:
> Why is this relevant? Won't tcpInfo.Notsent_bytes be zero if the connection is not established?

I don't get this comment. This check is not about connection establishment

> Also, we won't have the conenction object if it's not established.

This function should not be called when the connection is not yet established.

Reply:
I'm saying we can remove the `State` test. It's not doing anything. `tcpInfo.Notsent_bytes` will be zero if it's not established.

Reply:
Got it, thanks

Reply:
Hmm, what about the cases when TCP connection is closing? This way we can abort waiting loop earlier. Maybe that's why it was there in the original code.